homeassistant/components/recorder/statistics.py

```diff
@@ -423,6 +417,69 @@ def delete_duplicates(hass: HomeAssistant, session: Session) -> None:
         )
 
 
+def _find_statistics_meta_duplicates(session: Session) -> list[int]:
+    """Find duplicated statistics_meta."""
+    subquery = (
+        session.query(
+            StatisticsMeta.statistic_id,
+            literal_column("1").label("is_duplicate"),
```

Comment:
```diff
diff --git a/homeassistant/components/recorder/statistics.py b/homeassistant/components/recorder/statistics.py
index 270c1476a8..ed38dfd284 100644
--- a/homeassistant/components/recorder/statistics.py
+++ b/homeassistant/components/recorder/statistics.py
@@ -296,7 +296,6 @@ def _find_duplicates(
         session.query(
             table.start,
             table.metadata_id,
-            literal_column("1").label("is_duplicate"),
         )
         .group_by(table.metadata_id, table.start)
         .having(func.count() > 1)
@@ -309,7 +308,6 @@ def _find_duplicates(
             (subquery.c.metadata_id == table.metadata_id)
             & (subquery.c.start == table.start),
         )
-        .filter(subquery.c.is_duplicate == 1)
         .order_by(table.metadata_id, table.start, table.id.desc())
         .limit(1000 * MAX_ROWS_TO_PURGE)
     )
@@ -422,7 +420,6 @@ def _find_statistics_meta_duplicates(session: Session) -> list[int]:
     subquery = (
         session.query(
             StatisticsMeta.statistic_id,
-            literal_column("1").label("is_duplicate"),
         )
         .group_by(StatisticsMeta.statistic_id)
         .having(func.count() > 1)
@@ -434,7 +431,6 @@ def _find_statistics_meta_duplicates(session: Session) -> list[int]:
             subquery,
             (subquery.c.statistic_id == StatisticsMeta.statistic_id),
         )
-        .filter(subquery.c.is_duplicate == 1)
         .order_by(StatisticsMeta.statistic_id, StatisticsMeta.id.desc())
         .limit(1000 * MAX_ROWS_TO_PURGE)
     )
```

I think you can drop it since its already constrained by `.having(func.count() > 1)`

Reply:
Edit: but only if if you change the `outerjoin` to a `join`

Reply:
After the Discord discussion, I'd prefer not to change this since it seems the difference between an `outerjoin` and a `join` is very small and the version with `outerjoin` has been in production for some time.